Add guidance around Commit reveal and Immunity Period (#442)
opentensor/developer-docs: docs/subnets/commit-reveal.md

## Commit Reveal and Immunity Period

The [Immunity Period](../glossary.md#immunity-period) is the interval (measured in blocks) during which a miner or validator newly registered on a subnet is 'immune' from deregistration due to performance, because their performance is not yet scored. The duration of this period value should always be larger than the Commit Reveal interval, otherwise the immunity period will expire before a given miner's scores are available, and they may be deregistered without having their work counted.

To ensure that the immunity period is always greater than the Commit Reveal interval, we recommend using the following logic when setting the Immunity Period or Commit Reveal interval hyperparameters for a subnet:

```
new_immunity_period = (new_commit_reveal_interval - old_commit_reveal_interval) + old_immunity_period
```

See [Subnet Hyperparameters](./subnet-hyperparameters.md).

## Commit reveal in detail

When commit reveal is enabled, it works as follows:  

1. A subnet validator sets the weights normally by using [`set_weights`](pathname:///python-api/html/autoapi/bittensor/core/extrinsics/set_weights/index.html). 

> 2. Instead of publishing weights openly, an encrypted hash of these consensus weights is committed to the blockchain, using an internal method called [`commit_weights`](pathname:///python-api/html/autoapi/bittensor/core/extrinsics/commit_weights/index.html).

Reviewer: ```suggestion
2. Instead of publishing weights openly, an encrypted copy of these weights is committed to the blockchain, using an internal method called [`commit_weights`](pathname:///python-api/html/autoapi/bittensor/core/extrinsics/commit_weights/index.html).
```

3. A waiting interval, specified as a number of tempos, elapses. Subnet owners configure this interval with the subnet hyperparameter `commit_reveal_weights_interval`.

> 4. After this interval has elapsed, the unencrypted weights are published to the blockchain.

Author: Add note about ‘deren’ (?) pulse mechanism for publishing decryption key after the commit reveal interval that unlocks the previously published weight matrix.

> 4. After this interval has elapsed, the unencrypted weights are published to the blockchain.

Reviewer: ```suggestion
4. After this interval has elapsed, the unencrypted weights are automatically revealed by the chain.
```

Reviewer: Not sure if we need to explain why here, but basically the encryption is time-based using the drand network to provide the time-lock encryption.

Validators specify a time to reveal the weights and encrypt them, then publish the encrypted weights for later automatic reveal.

> 5. On the blockchain, the committed hash from step 2 is compared to the hash of the openly submitted (revealed) weights from step 4:

Reviewer: This can be removed now. See above comment.

/>
</center>

## How to use the commit reveal feature

> As a subnet owner, set the below hyperparameters to use the commit reveal feature:

Reviewer: Might be good to specify what minimum version of the python API needs to be used to support this feature.
